src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -41,7 +42,7 @@ public Book update(@RequestBody Book book, @PathVariable Long id){
     }
 
     @DeleteMapping("/{id}")
-    public Book delete(@PathVariable Long id){
+    public Book delete(@PathVariable Long id) throws BookNotFoundException {
         if(bookRepository.findById(id) != null){
```

Comment:
Now that you've got your exception and that findById returns an Optional, you could rewrite this as:
bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
bookrepository.deleteById(id);